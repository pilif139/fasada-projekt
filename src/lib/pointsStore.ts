import { writable } from "svelte/store";

export const points = writable(0);