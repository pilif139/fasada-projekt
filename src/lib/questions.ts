export type QuestionType = {
    pageId: number;
    question: string;
    answers: string[];
    correctAnswer: string | number | boolean | string[];
}

export const questions : QuestionType[] = [
    {
        pageId: 1,
        question: "What's your name?",
        answers: [
            "Filip",
            "John",
            "Arthur",
        ],
        correctAnswer: 0, // index of correct answer in answers array
    },
    {
        pageId: 1,
        question: "What's your age?",
        answers: [
            "20",
            "30",
            "40",
        ],
        correctAnswer: 1,
    },
];