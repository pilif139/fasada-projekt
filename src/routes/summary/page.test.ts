import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/svelte';
import SummaryPage from './+page.svelte';
import { points } from '$lib/pointsStore';
import { questions } from '$lib/questions';
import '@testing-library/jest-dom/vitest';

describe('Summary Page Component', () => {
  it('renders summary page matching with snapshot', () => {
    const {container} = render(SummaryPage);
    expect(container).toMatchSnapshot();
  });

  it('displays total points earned', () => {
    points.set({ points: 4, YourAnswers: [0,1,0,1] });
    render(SummaryPage);
    expect(screen.getByText('4')).toBeInTheDocument();
  });

  it('renders each question with correct answer', () => {
    points.set({ points: 10, YourAnswers: [0, 1, 2, 2] });
    render(SummaryPage);
    questions.forEach((question) => {
      expect(screen.getByText(question.question)).toBeInTheDocument();
      expect(screen.getByText(question.answers[question.correctAnswer])).toBeInTheDocument();
    });
  });
});