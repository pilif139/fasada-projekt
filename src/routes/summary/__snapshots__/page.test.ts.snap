// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Summary Page Component > renders summary page matching with snapshot 1`] = `
<div>
  <main
    class="flex flex-col items-center text-2xl font-bold gap-4 w-[80vw] md:w-[40vw] my-4"
  >
    <p
      class="text-5xl font-extrabold"
    >
      Podsumowanie
    </p>
     
    <p>
      Ilość punktów: 
      <span
        class="font-extrabold text-blue-500"
      >
        0
      </span>
    </p>
     
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Co robi fasada?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: Przedstawia skomplikowany kod w postaci prostego interfejsu
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Ile fasad możemy mieć w programie?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: ile chcemy
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Jaka fasada przekierowuje użytkownika do kolejnych?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: fasada główna
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Jak powinna wyglądać główna fasada?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: powinna być przejrztysta
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Czym jest podsystem?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: klasami i obiektami
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Która z tych cech NIE jest zaletą fasady?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: możliwość ograniczenia liczby użytkowników programu
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Która z tych cech jest wadą fasady?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: Wydłużenie czasu tworzenie kodu
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Kiedy dzwonisz, aby złożyć zamówienie, co lub kto pełni funkcję fasady?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: pracownik odbierający telefon
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Jakiej funkcji NIE pełni fasada w aplikacji bankomatowej?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: obliczania aktualnego stanu konta bankowego
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="p-10 font-bold text-white rounded-xl transition-all duration-500 bg-gradient-to-t from-slate-900 via-blue-400 to-blue-600  bg-size-200 bg-pos-0 hover:bg-pos-100 my-5 cursor-pointer w-full shadow-lg shadow-black from-slate-950 via-blue-950 to-gray-900 min-w-[45vw]"
    >
      <p
        class="text-3xl font-extrabold"
      >
        Jaki element kalkulatora stanowi część fasady?
      </p>
       
      <div
        class="flex flex-wrap gap-2"
      >
        <p>
          Poprawna odpowiedź: przyciski
        </p>
         
        <p
          aria-label="wrong-answer"
          class="text-red-500"
        >
          Twoja odpowiedź: 
        </p>
        <!---->
      </div>
      <!---->
    </div>
    <!---->
     
    <a
      class="bg-blue-500 px-8 py-4 rounded-xl text-xl font-bold hover:bg-blue-800 transition-all border-2 border-blue-600 hover:scale-110 duration-300 "
      href="/"
    >
      Rozwiąż jeszcze raz
      <!---->
    </a>
    <!---->
  </main>
  
</div>
`;
